### PR TITLE
feat(api): add mock for `convertFileSrc`, closes #7935

### LINKED ISSUE #7935: [bug] window is not defined when using 'convertFileSrc' in unit test
### Describe the bug

I can't find anything about it in the changelog, but starting from tauri v1.5.0, I get `ReferenceError: window is not defined` when using `convertFileSrc` in a unit test. 

### Reproduction

1) Create unit test - I used vitest
2) Call function from my own code that contains `convertFileSrc` and test is inside the unit test

### Expected behavior

I would expect that this still works as it did `v1.3.0` and `v1.4.0`

### Platform and versions

```text
pnpm tauri info

[✔] Environment
    - OS: Mac OS 14.0.0 X64
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.72.1 (d5c2e9c34 2023-09-13)
    ✔ Cargo: 1.72.1 (103a7ff2e 2023-08-15)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-aarch64-apple-darwin (default)
    - node: 18.16.0
    - pnpm: 8.8.0
    - npm: 9.5.1

[-] Packages
    - tauri [RUST]: 1.3.0
    - tauri-build [RUST]: 1.3.0
    - wry [RUST]: 0.24.3
    - tao [RUST]: 0.16.1
    - tauri-cli [RUST]: not installed!
    - @tauri-apps/api [NPM]: 1.5.0
    - @tauri-apps/cli [NPM]: 1.5.0

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../.output/public
    - devPath: http://localhost:8081/
    - framework: Vue.js (Nuxt)
    - bundler: Vite
```
```


### Stack trace

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by amrbashir**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [ ] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: tooling/api/src/mocks.ts [hunk 1]
```diff
@@ -142,6 +142,37 @@ export function mockWindows(
...
+ * @param osName The operating system to mock, can be one of linux, macos, or windows
+ * @param windowsProtocolScheme The scheme to use on Windows, can be either `http` or `https` and defaults to `https`
+ *
+ * @since 1.6.0
+ */
+export function mockConvertFileSrc(
+  osName: string,
+  windowsProtocolScheme = 'https'
+): void {
+  window.__TAURI__.convertFileSrc = function (filePath, protocol = 'asset') {
```
Comment (amrbashir) [inline tooling/api/src/mocks.ts:169]:
```suggestion
  window.__TAURI__ = window.__TAURI__ ?? {};
  window.__TAURI__.convertFileSrc = function (filePath, protocol = 'asset') {
```

**Review by FabianLars — APPROVED** [review]

File: .changes/api-convert-file-src-mock.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@tauri-apps/api': 'minor:feat'
```
Comment (lucasfernog) [inline .changes/api-convert-file-src-mock.md:2]: @amrbashir can we change this to a patch so we can release the Debian package hotfix?

Comment (amrbashir) [reply .changes/api-convert-file-src-mock.md:2]: we've got to have selective release in our workflow, this is not scalable IMO, but I will revert this bump.

Comment (amrbashir) [reply .changes/api-convert-file-src-mock.md:2]: https://github.com/tauri-apps/tauri/pull/8033

Comment (lucasfernog) [reply .changes/api-convert-file-src-mock.md:2]: yeah we should have that :( it's always so boring.

**Pull request merged**